docs/DeveloperGuide.md

```diff
@@ -1,34 +1,1080 @@
+
+In this example, it fetches the following `Anime` objects.
+```text
+Charlie
+Echo
+Gamma
+```
+
+If the 2nd page of the list was requested instead with the command `browse -p 2`. 
+`BrowseCommand#buildBrowseOutput()` will shift its page window down by 1 page as depicted in the diagram below.
+
+![Browse Object Diagram 2](images/Browse-Default-State2.png) <br/>
+*Figure 11: Browse Next Page Object Diagram*
+
+**Step 5:** At each `Anime` object, it will access its methods to get the relevant information about that anime series and construct a printable result for the user to view.
+
+**Step 6:** After all `Anime` objects in the page window have been retrieved, it will return the printable result back to `BrowseCommand#execute()`. At this point it will utilise `BrowseCommand#sortBrowseOption()` again to reset the `AnimeData` list to its original form if it has been altered.
+
+**Step 7:** Once `AnimeData` has been sorted, it will return the result back to `Main` for printing.
+
+Currently, this result is not very exciting as it is just based on the Anime ID which is assigned randomly by our data source, AniList.
+
+An example scenario would be browsing the 2nd page of a **sorted** list in ascending order.
+The only step that would change would be at Step 3, where it will perform sorting of `AnimeData` list. 
+
+![Browse Object Diagram 3](images/Browse-Sorted-State.png) <br/>
+*Figure 12: Browse Sorted State Object Diagram*
+
+As you can see, even though the page window is at the same position as the previous command, 
+the list is different as it has been sorted.
+
+From this point onwards, the operation will continue as per the steps above but during the final `BrowseCommand#sortBrowseList()` call in step 6. It will perform a sort to reset the list.
+
+Here is the sequence diagram to better illustrate the lifecycle of a browse command.
+
+![Browse Sequence Diagram](images/Browse-SequenceDiagram.png) <br/>
+*Figure 13: Browse Sorted Sequence Diagram*
+
+<br/>
+
+#### 4.2.2 Design Consideration
+Here are some various design considerations that was taken when implementing the `browse` feature.
+
+Aspect: **How should the program handle the sorted list**
+
+The first design consideration was how the sorting should be carried out. The main issues here are the time and storage complexity.
+
+| Approach | Pros | Cons  |
+| --- | --- | --- |
+| 1. Leaving the list unsorted                    | - No complexity and fastest approach  | - List will be unsorted and may cause confusion to users |
+| 2. Resorting the list again                     | - The list will be back into its original form before browsing    | - May hinder performance as resorting could take time <br/> - Requires altering of the main list |
+| 3. Cloning a duplicate animeData object to sort | - The list will be back to its original form <br/> - The main list will not be affected at all | - Expensive operation that will require large storage and time complexity |
```

Comment:
![image](https://user-images.githubusercontent.com/60333990/97657199-3a4cc300-1aa4-11eb-8c22-15b9e6409865.png)
The use of a table is appropriate but it is a little wordy. Perhaps some icons could be used to increase the readability of the table.